static/scripts/onboarding/github-login-button.ts

```diff
@@ -0,0 +1,124 @@
+import { createClient } from "@supabase/supabase-js";
+import { getLocalStore } from "./get-local-store";
+
+declare const SUPABASE_URL: string;
+declare const SUPABASE_ANON_KEY: string;
+
+const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);
+
+export interface OAuthToken {
+  provider_token: string;
+  access_token: string;
+  expires_in: number;
+  expires_at: number;
+  refresh_token: string;
+  token_type: string;
+  user: {
+    id: string;
+    aud: string;
+    role: string;
+    email: string;
+    email_confirmed_at: string;
+    phone: string;
+    confirmed_at: string;
+    last_sign_in_at: string;
+    app_metadata: { provider: string; providers: string[] };
+    user_metadata: {
+      avatar_url: string;
+      email: string;
+      email_verified: boolean;
+      full_name: string;
+      iss: string;
+      name: string;
+      phone_verified: boolean;
+      preferred_username: string;
+      provider_id: string;
+      sub: string;
+      user_name: string;
+    };
+    identities: [
+      {
+        id: string;
+        user_id: string;
+        identity_data: {
+          avatar_url: string;
+          email: string;
+          email_verified: boolean;
+          full_name: string;
+          iss: string;
+          name: string;
+          phone_verified: boolean;
+          preferred_username: string;
+          provider_id: string;
+          sub: string;
+          user_name: string;
+        };
+        provider: string;
+        last_sign_in_at: string;
+        created_at: string;
+        updated_at: string;
+      },
+    ];
+    created_at: string;
+    updated_at: string;
+  };
+}
+
+declare const FRONTEND_URL: string;
+
+async function gitHubLoginButtonHandler() {
+  const { error } = await supabase.auth.signInWithOAuth({
+    provider: "github",
+    options: {
+      redirectTo: FRONTEND_URL,
+      scopes: "admin:org user:read repo",
+    },
+  });
+  if (error) {
+    console.error("Error logging in:", error);
+  }
+}
+const gitHubLoginButton = document.createElement("button");
+export async function renderGitHubLoginButton() {
+  const stepContainer = document.getElementById("overlay-item-container");
+  const overlay = document.getElementById("overlay");
+  const setBtn = document.getElementById("setBtn");
+
+  // No need to show the OAuth button if we are already logged in
+  if (getSessionToken()) {
+    overlay?.classList.add("hidden");
+    return;
+  }
+
+  gitHubLoginButton.id = "github-login-button";
+  gitHubLoginButton.innerHTML = "<span>Connect</span><span class='full'>&nbsp;To GitHub</span>";
+  gitHubLoginButton.addEventListener("click", gitHubLoginButtonHandler);
+  if (stepContainer) {
+    stepContainer.insertBefore(gitHubLoginButton, stepContainer.firstChild);
+  }
+  setBtn?.setAttribute("disabled", "");
```

Comment:
```suggestion
  setButton.disabled = false
```

Is the correct way to do this. `setButton` should be casted as `HTMLButtonElement`